File: backend-v2/schemas_new/tracking.py
"""
Pydantic schemas for conversion tracking and attribution.
"""

from pydantic import BaseModel, Field, validator, field_validator
from typing import Dict, List, Optional, Any
from datetime import datetime
from enum import Enum
import re

from models.tracking import EventType, AttributionModel, ConversionStatus


class ConversionEventCreate(BaseModel):
    """Schema for creating a new conversion event"""
    event_id: Optional[str] = Field(None, description="Unique event ID for deduplication")
    event_name: str = Field(..., description="Name of the event (e.g., 'booking_completed')")
    event_type: EventType = Field(..., description="Type of conversion event")
    event_value: Optional[float] = Field(None, description="Monetary value of the event")
    event_currency: Optional[str] = Field("USD", description="Currency code")
    event_data: Optional[Dict[str, Any]] = Field(None, description="Additional event data")
    
    # Source information
    source_url: Optional[str] = Field(None, description="URL where event occurred")
    user_agent: Optional[str] = Field(None, description="User agent string")
    ip_address: Optional[str] = Field(None, description="User IP address (will be hashed)")
    client_id: Optional[str] = Field(None, description="Google Analytics client ID")
    session_id: Optional[str] = Field(None, description="Session identifier")
    
    # Attribution data
    utm_source: Optional[str] = Field(None, description="UTM source parameter")
    utm_medium: Optional[str] = Field(None, description="UTM medium parameter")
    utm_campaign: Optional[str] = Field(None, description="UTM campaign parameter")
    utm_term: Optional[str] = Field(None, description="UTM term parameter")
    utm_content: Optional[str] = Field(None, description="UTM content parameter")
    referrer: Optional[str] = Field(None, description="HTTP referrer")
    
    @validator('event_value')
    def validate_event_value(cls, v):
        if v is not None and v < 0:
            raise ValueError('Event value must be non-negative')
        return v
    
    @validator('event_currency')
    def validate_currency(cls, v):
        if v and len(v) != 3:
            raise ValueError('Currency must be a 3-letter code')
        return v.upper() if v else 'USD'


class ConversionEventResponse(BaseModel):
    """Schema for conversion event response"""
    id: int
    user_id: int
    event_id: str
    event_name: str
    event_type: EventType
    event_value: Optional[float]
    event_currency: str
    channel: Optional[str]
    status: ConversionStatus
    created_at: datetime
    
    # Sync status
    gtm_synced: bool
    meta_synced: bool
    google_ads_synced: bool
    
    # Attribution
    attribution_path_id: Optional[int] = None
    
    class Config:
        from_attributes = True


class AttributionReport(BaseModel):
    """Schema for attribution reporting"""
    model: AttributionModel
    total_revenue: float
    total_conversions: int
    channels: List[Dict[str, Any]]
    period_start: datetime
    period_end: datetime


class ChannelPerformance(BaseModel):
    """Schema for channel performance metrics"""
    channel: str
    conversions: int
    revenue: float
    attributed_revenue: float
    conversion_rate: float
    roi: float


class ConversionAnalytics(BaseModel):
    """Schema for conversion analytics response"""
    total_conversions: int
    total_revenue: float
    conversion_rate: float
    average_order_value: float
    channel_performance: List[ChannelPerformance]
    top_converting_pages: List[Dict[str, Any]]
    conversion_funnel: List[Dict[str, Any]]
    period_start: datetime
    period_end: datetime


class TrackingConfigUpdate(BaseModel):
    """Schema for updating tracking configuration"""
    # GTM settings
    gtm_container_id: Optional[str] = None
    gtm_enabled: Optional[bool] = None
    gtm_server_url: Optional[str] = None
    
    # Meta settings
    meta_pixel_id: Optional[str] = None
    meta_enabled: Optional[bool] = None
    meta_test_event_code: Optional[str] = None
    
    # Google Ads settings
    google_ads_conversion_id: Optional[str] = None
    google_ads_enabled: Optional[bool] = None
    google_ads_conversion_labels: Optional[Dict[str, str]] = None
    
    # Attribution settings
    attribution_window_days: Optional[int] = Field(None, ge=1, le=90)
    default_attribution_model: Optional[AttributionModel] = None
    
    # Advanced settings
    conversion_value_rules: Optional[Dict[str, Any]] = None
    excluded_domains: Optional[List[str]] = None


class TrackingConfigResponse(BaseModel):
    """Schema for tracking configuration response"""
    id: int
    user_id: int
    
    # Platform settings
    gtm_enabled: bool
    gtm_container_id: Optional[str]
    meta_enabled: bool
    meta_pixel_id: Optional[str]
    google_ads_enabled: bool
    google_ads_conversion_id: Optional[str]
    
    # Attribution settings
    attribution_window_days: int
    default_attribution_model: AttributionModel
    
    # Privacy settings
    enable_enhanced_conversions: bool
    hash_user_data: bool
    
    created_at: datetime
    updated_at: Optional[datetime]
    
    class Config:
        from_attributes = True


class ConversionGoalCreate(BaseModel):
    """Schema for creating a conversion goal"""
    name: str = Field(..., description="Goal name")
    description: Optional[str] = Field(None, description="Goal description")
    event_name: str = Field(..., description="Event that triggers this goal")
    event_type: EventType = Field(..., description="Type of event")
    value: Optional[float] = Field(None, description="Fixed goal value")
    value_expression: Optional[str] = Field(None, description="Dynamic value calculation")
    conditions: Optional[Dict[str, Any]] = Field(None, description="Additional conditions")
    is_active: bool = Field(True, description="Whether goal is active")


class ConversionGoalResponse(BaseModel):
    """Schema for conversion goal response"""
    id: int
    user_id: int
    name: str
    description: Optional[str]
    event_name: str
    event_type: EventType
    value: Optional[float]
    value_expression: Optional[str]
    conditions: Optional[Dict[str, Any]]
    is_active: bool
    total_conversions: int
    total_value: float
    created_at: datetime
    updated_at: Optional[datetime]
    
    class Config:
        from_attributes = True


class CampaignTrackingCreate(BaseModel):
    """Schema for creating campaign tracking"""
    campaign_id: str = Field(..., description="External campaign ID")
    campaign_name: str = Field(..., description="Campaign name")
    campaign_source: str = Field(..., description="Campaign source (google_ads, meta, etc.)")
    campaign_medium: str = Field(..., description="Campaign medium")
    start_date: datetime = Field(..., description="Campaign start date")
    end_date: Optional[datetime] = Field(None, description="Campaign end date")
    total_cost: float = Field(0.0, description="Total campaign cost")
    currency: str = Field("USD", description="Currency code")


class CampaignTrackingResponse(BaseModel):
    """Schema for campaign tracking response"""
    id: int
    user_id: int
    campaign_id: str
    campaign_name: str
    campaign_source: str
    campaign_medium: str
    start_date: datetime
    end_date: Optional[datetime]
    total_cost: float
    currency: str
    
    # Performance metrics
    impressions: int
    clicks: int
    conversions: int
    conversion_value: float
    
    # Calculated metrics
    ctr: Optional[float]
    conversion_rate: Optional[float]
    cpc: Optional[float]
    cpa: Optional[float]
    roas: Optional[float]
    
    is_active: bool
    created_at: datetime
    updated_at: Optional[datetime]
    last_sync_at: Optional[datetime]
    
    class Config:
        from_attributes = True


class PlatformTestRequest(BaseModel):
    """Schema for testing platform connections"""
    platform: str = Field(..., description="Platform to test (gtm, meta, google_ads)")
    config: Dict[str, Any] = Field(..., description="Platform-specific configuration")


class PlatformTestResponse(BaseModel):
    """Schema for platform test response"""
    success: bool
    message: str
    details: Optional[Dict[str, Any]] = None


# Customer Pixel Management Schemas

class TrackingPixelUpdate(BaseModel):
    """Schema for updating organization tracking pixels"""
    
    gtm_container_id: Optional[str] = Field(None, max_length=50, description="Google Tag Manager Container ID (GTM-XXXXXXX)")
    ga4_measurement_id: Optional[str] = Field(None, max_length=50, description="Google Analytics 4 Measurement ID (G-XXXXXXXXXX)")
    meta_pixel_id: Optional[str] = Field(None, max_length=50, description="Meta/Facebook Pixel ID")
    google_ads_conversion_id: Optional[str] = Field(None, max_length=50, description="Google Ads Conversion ID (AW-XXXXXXXXX)")
    google_ads_conversion_label: Optional[str] = Field(None, max_length=50, description="Google Ads Conversion Label")
    tracking_enabled: Optional[bool] = Field(None, description="Enable/disable all tracking pixels")
    custom_tracking_code: Optional[str] = Field(None, description="Custom HTML/JS tracking code")
    tracking_settings: Optional[Dict[str, Any]] = Field(None, description="Advanced tracking settings")
    
    @field_validator('gtm_container_id')
    def validate_gtm_id(cls, v):
        if v and not re.match(r'^GTM-[A-Z0-9]{6,}$', v):
            raise ValueError('Invalid GTM Container ID format. Must be GTM-XXXXXXX')
        return v
    
    @field_validator('ga4_measurement_id')
    def validate_ga4_id(cls, v):
        if v and not re.match(r'^G-[A-Z0-9]{10,}$', v):
            raise ValueError('Invalid GA4 Measurement ID format. Must be G-XXXXXXXXXX')
        return v
    
    @field_validator('meta_pixel_id')
    def validate_meta_pixel_id(cls, v):
        if v and not re.match(r'^\d{10,20}$', v):
            raise ValueError('Invalid Meta Pixel ID format. Must be numeric')
        return v
    
    @field_validator('google_ads_conversion_id')
    def validate_google_ads_id(cls, v):
        if v and not re.match(r'^AW-\d{9,}$', v):
            raise ValueError('Invalid Google Ads Conversion ID format. Must be AW-XXXXXXXXX')
        return v
    
    @field_validator('custom_tracking_code')
    def validate_custom_code(cls, v):
        if v:
            # Basic security check - no script tags with src
            if re.search(r'<script[^>]+src\s*=', v, re.IGNORECASE):
                raise ValueError('External script sources not allowed in custom tracking code')
            # Limit length
            if len(v) > 10000:
                raise ValueError('Custom tracking code too long (max 10000 characters)')
        return v
    
    class Config:
        json_schema_extra = {
            "example": {
                "gtm_container_id": "GTM-ABC123",
                "ga4_measurement_id": "G-1234567890",
                "meta_pixel_id": "123456789012345",
                "google_ads_conversion_id": "AW-123456789",
                "google_ads_conversion_label": "abcDEFghijk",
                "tracking_enabled": True,
                "tracking_settings": {
                    "enhanced_conversions": True,
                    "cookieless_tracking": False
                }
            }
        }


class TrackingPixelResponse(BaseModel):
    """Schema for tracking pixel response"""
    
    gtm_container_id: Optional[str] = None
    ga4_measurement_id: Optional[str] = None
    meta_pixel_id: Optional[str] = None
    google_ads_conversion_id: Optional[str] = None
    google_ads_conversion_label: Optional[str] = None
    tracking_enabled: bool = True
    custom_tracking_code: Optional[str] = None
    tracking_settings: Optional[Dict[str, Any]] = None
    
    class Config:
        from_attributes = True


class TrackingTestResult(BaseModel):
    """Schema for pixel test results"""
    
    pixel_type: str = Field(..., description="Type of pixel tested (gtm, ga4, meta, google_ads)")
    is_valid: bool = Field(..., description="Whether the pixel ID is valid")
    is_active: bool = Field(..., description="Whether the pixel is firing correctly")
    message: str = Field(..., description="Test result message")
    details: Optional[Dict[str, Any]] = Field(None, description="Additional test details")
    
    class Config:
        json_schema_extra = {
            "example": {
                "pixel_type": "gtm",
                "is_valid": True,
                "is_active": True,
                "message": "GTM container loaded successfully",
                "details": {
                    "container_version": "2",
                    "tags_count": 5
                }
            }
        }


class PublicTrackingPixels(BaseModel):
    """Schema for public-facing tracking pixels (used on booking pages)"""
    
    gtm_container_id: Optional[str] = None
    ga4_measurement_id: Optional[str] = None
    meta_pixel_id: Optional[str] = None
    google_ads_conversion_id: Optional[str] = None
    google_ads_conversion_label: Optional[str] = None
    custom_tracking_code: Optional[str] = None
    tracking_enabled: bool = True
    
    class Config:
        from_attributes = True


class PixelInstructions(BaseModel):
    """Schema for pixel setup instructions"""
    
    pixel_type: str = Field(..., description="Type of pixel (gtm, ga4, meta, google_ads)")
    title: str = Field(..., description="Instruction title")
    description: str = Field(..., description="Brief description of the pixel")
    setup_steps: List[str] = Field(..., description="Step-by-step setup instructions")
    format_example: str = Field(..., description="Example of correct ID format")
    validation_regex: str = Field(..., description="Regex pattern for validation")
    help_url: Optional[str] = Field(None, description="URL to official documentation")
    
    class Config:
        json_schema_extra = {
            "example": {
                "pixel_type": "gtm",
                "title": "Google Tag Manager Setup",
                "description": "Track all website events with GTM",
                "setup_steps": [
                    "Go to tagmanager.google.com",
                    "Create or select a container",
                    "Copy your Container ID"
                ],
                "format_example": "GTM-ABC123",
                "validation_regex": "^GTM-[A-Z0-9]{6,}$",
                "help_url": "https://support.google.com/tagmanager/answer/6103696"
            }
        }

File: backend-v2/frontend-v2/types/tracking.ts
// Conversion tracking type definitions

// Define ConversionEventType enum
export enum ConversionEventType {
  PAGE_VIEW = 'page_view',
  VIEW_CONTENT = 'view_content',
  SEARCH = 'search',
  ADD_TO_CART = 'add_to_cart',
  ADD_TO_WISHLIST = 'add_to_wishlist',
  INITIATE_CHECKOUT = 'begin_checkout',
  ADD_PAYMENT_INFO = 'add_payment_info',
  PURCHASE = 'purchase',
  LEAD = 'lead',
  COMPLETE_REGISTRATION = 'complete_registration',
  CONTACT = 'contact',
  SUBMIT_APPLICATION = 'submit_application',
  SUBSCRIBE = 'subscribe',
  START_TRIAL = 'start_trial',
  SCHEDULE = 'schedule'
}

// Define ConversionEventParams interface
export interface ConversionEventParams {
  value?: number
  currency?: string
  content_name?: string
  content_category?: string
  content_ids?: string[]
  contents?: Array<{ id: string; quantity: number }>
  content_type?: string
  num_items?: number
}

// Define PurchaseEventParams interface
export interface PurchaseEventParams extends ConversionEventParams {
  transaction_id: string
  affiliation?: string
  revenue?: number
  tax?: number
  shipping?: number
  coupon?: string
}

// Event Type Enum (matching backend)
export enum EventType {
  PAGE_VIEW = "page_view",
  CLICK = "click",
  FORM_SUBMIT = "form_submit",
  ADD_TO_CART = "add_to_cart",
  PURCHASE = "purchase",
  REGISTRATION = "registration",
  LEAD = "lead",
  PHONE_CALL = "phone_call",
  CHAT_STARTED = "chat_started",
  CUSTOM = "custom"
}

// Attribution Model Enum (matching backend)
export enum AttributionModel {
  LAST_CLICK = "last_click",
  FIRST_CLICK = "first_click",
  LINEAR = "linear",
  TIME_DECAY = "time_decay",
  POSITION_BASED = "position_based",
  DATA_DRIVEN = "data_driven"
}

// Conversion Status Enum (matching backend)
export enum ConversionStatus {
  PENDING = "pending",
  TRACKED = "tracked",
  FAILED = "failed",
  DUPLICATE = "duplicate"
}

// API Request/Response Types
export interface ConversionEventCreate {
  event_id?: string
  event_name: string
  event_type: EventType
  event_value?: number
  event_currency?: string
  event_data?: Record<string, any>
  source_url?: string
  user_agent?: string
  ip_address?: string
  client_id?: string
  session_id?: string
  utm_source?: string
  utm_medium?: string
  utm_campaign?: string
  utm_term?: string
  utm_content?: string
  referrer?: string
}

export interface ConversionEventResponse {
  id: number
  user_id: number
  event_id: string
  event_name: string
  event_type: EventType
  event_value?: number
  event_currency: string
  channel?: string
  status: ConversionStatus
  created_at: string
  gtm_synced: boolean
  meta_synced: boolean
  google_ads_synced: boolean
  attribution_path_id?: number
}

export interface ChannelPerformance {
  channel: string
  conversions: number
  revenue: number
  attributed_revenue: number
  conversion_rate: number
  roi: number
}

export interface ConversionAnalytics {
  total_conversions: number
  total_revenue: number
  conversion_rate: number
  average_order_value: number
  channel_performance: ChannelPerformance[]
  top_converting_pages: Record<string, any>[]
  conversion_funnel: Record<string, any>[]
  period_start: string
  period_end: string
}

export interface AttributionReport {
  model: AttributionModel
  total_revenue: number
  total_conversions: number
  channels: Record<string, any>[]
  period_start: string
  period_end: string
}

export interface TrackingConfigUpdate {
  gtm_container_id?: string
  gtm_enabled?: boolean
  gtm_server_url?: string
  meta_pixel_id?: string
  meta_enabled?: boolean
  meta_test_event_code?: string
  google_ads_conversion_id?: string
  google_ads_enabled?: boolean
  google_ads_conversion_labels?: Record<string, string>
  attribution_window_days?: number
  default_attribution_model?: AttributionModel
  conversion_value_rules?: Record<string, any>
  excluded_domains?: string[]
}

export interface TrackingConfigResponse {
  id: number
  user_id: number
  gtm_enabled: boolean
  gtm_container_id?: string
  meta_enabled: boolean
  meta_pixel_id?: string
  google_ads_enabled: boolean
  google_ads_conversion_id?: string
  attribution_window_days: number
  default_attribution_model: AttributionModel
  enable_enhanced_conversions: boolean
  hash_user_data: boolean
  created_at: string
  updated_at?: string
}

export interface ConversionGoalCreate {
  name: string
  description?: string
  event_name: string
  event_type: EventType
  value?: number
  value_expression?: string
  conditions?: Record<string, any>
  is_active?: boolean
}

export interface ConversionGoalResponse {
  id: number
  user_id: number
  name: string
  description?: string
  event_name: string
  event_type: EventType
  value?: number
  value_expression?: string
  conditions?: Record<string, any>
  is_active: boolean
  total_conversions: number
  total_value: number
  created_at: string
  updated_at?: string
}

export interface CampaignTrackingCreate {
  campaign_id: string
  campaign_name: string
  campaign_source: string
  campaign_medium: string
  start_date: string
  end_date?: string
  total_cost?: number
  currency?: string
}

export interface CampaignTrackingResponse {
  id: number
  user_id: number
  campaign_id: string
  campaign_name: string
  campaign_source: string
  campaign_medium: string
  start_date: string
  end_date?: string
  total_cost: number
  currency: string
  impressions: number
  clicks: number
  conversions: number
  conversion_value: number
  ctr?: number
  conversion_rate?: number
  cpc?: number
  cpa?: number
  roas?: number
  is_active: boolean
  created_at: string
  updated_at?: string
  last_sync_at?: string
}

export interface CampaignMetricsUpdate {
  impressions?: number
  clicks?: number
  conversions?: number
  conversion_value?: number
  total_cost?: number
}

export interface PlatformTestRequest {
  platform: 'gtm' | 'meta' | 'google_ads'
  config: Record<string, any>
}

export interface PlatformTestResponse {
  success: boolean
  message: string
  details?: Record<string, any>
}

export interface TrackingHealthStatus {
  status: 'healthy' | 'degraded' | 'unhealthy'
  integrations: {
    gtm: {
      configured: boolean
      enabled: boolean
    }
    meta: {
      configured: boolean
      enabled: boolean
    }
    google_ads: {
      configured: boolean
      enabled: boolean
    }
  }
  events_last_24h: number
  timestamp: string
}

export interface CampaignSyncResponse {
  synced: number
  errors: Array<{
    campaign_id: string
    source: string
    error: string
  }>
  message: string
}

// Query Parameters
export interface ConversionAnalyticsQuery {
  start_date?: string
  end_date?: string
  group_by?: 'day' | 'week' | 'month'
}

export interface AttributionReportQuery {
  model?: AttributionModel
  start_date?: string
  end_date?: string
}

export interface ConversionGoalsQuery {
  is_active?: boolean
}

export interface CampaignTrackingQuery {
  is_active?: boolean
  source?: string
  start_date?: string
  end_date?: string
}

export interface CampaignSyncQuery {
  source?: string
}

// Extended event parameters for specific use cases
export interface BookingEventParams extends ConversionEventParams {
  barber_id?: string
  barber_name?: string
  location_id?: string
  location_name?: string
  appointment_date?: string
  appointment_time?: string
  booking_type?: 'walk-in' | 'appointment' | 'recurring'
  booking_source?: 'web' | 'mobile' | 'phone' | 'in-person'
  services?: string[]
}

export interface UserEventParams extends ConversionEventParams {
  method?: 'email' | 'google' | 'facebook' | 'apple'
  user_role?: 'customer' | 'barber' | 'admin' | 'manager'
  referral_code?: string
  utm_source?: string
  utm_medium?: string
  utm_campaign?: string
}

export interface ReviewEventParams extends ConversionEventParams {
  rating: number
  barber_id: string
  service_ids: string[]
  has_text: boolean
  has_photos: boolean
  days_after_appointment: number
  platform?: 'internal' | 'google' | 'facebook' | 'yelp'
}

export interface MarketingEventParams extends ConversionEventParams {
  campaign_id?: string
  campaign_name?: string
  campaign_type?: 'email' | 'sms' | 'push' | 'social'
  message_id?: string
  variant?: string
  action?: 'opened' | 'clicked' | 'converted'
}

// Conversion value helpers
export interface ConversionValue {
  value: number
  currency: string
  ltv?: number // Lifetime value
  average_order_value?: number
  purchase_frequency?: number
}

// Event context for enhanced tracking
export interface EventContext {
  session_id?: string
  client_id?: string
  user_agent?: string
  ip_country?: string
  device_type?: 'desktop' | 'mobile' | 'tablet'
  browser?: string
  os?: string
  referrer?: string
  landing_page?: string
  exit_page?: string
  pages_viewed?: number
  time_on_site?: number
}

// Booking funnel stages
export enum BookingFunnelStage {
  LANDING = 'landing',
  SERVICE_BROWSING = 'service_browsing',
  SERVICE_SELECTED = 'service_selected',
  BARBER_SELECTED = 'barber_selected',
  TIME_SELECTED = 'time_selected',
  CONTACT_INFO = 'contact_info',
  PAYMENT_INFO = 'payment_info',
  CONFIRMATION = 'confirmation',
  COMPLETED = 'completed',
}

// Attribution models
export interface AttributionData {
  first_touch?: {
    source: string
    medium: string
    campaign?: string
    timestamp: string
  }
  last_touch?: {
    source: string
    medium: string
    campaign?: string
    timestamp: string
  }
  multi_touch?: Array<{
    source: string
    medium: string
    campaign?: string
    timestamp: string
    credit: number // Attribution credit (0-1)
  }>
}

// Custom dimension mappings for GA4
export interface CustomDimensions {
  user_type?: string
  barber_id?: string
  location_id?: string
  service_category?: string
  booking_source?: string
  loyalty_tier?: string
  lifetime_value_tier?: string
  days_since_last_visit?: number
  total_bookings?: number
  preferred_barber?: string
}

// Custom metrics for GA4
export interface CustomMetrics {
  booking_value?: number
  service_duration?: number
  days_until_appointment?: number
  cancellation_rate?: number
  no_show_rate?: number
  rebooking_rate?: number
  average_rating?: number
  tip_amount?: number
  discount_amount?: number
  net_revenue?: number
}

// Enhanced e-commerce data for detailed tracking
export interface EnhancedEcommerceData extends PurchaseEventParams {
  affiliation?: string // Shop/location name
  revenue?: number // Total revenue after discounts
  shipping?: number
  tax?: number
  discount?: number
  coupon?: string
  list_name?: string // Where the service was selected from
  list_id?: string
  items: Array<{
    item_id: string
    item_name: string
    affiliation?: string
    coupon?: string
    discount?: number
    index?: number // Position in list
    item_brand?: string // Barber name
    item_category?: string
    item_category2?: string
    item_category3?: string
    item_list_id?: string
    item_list_name?: string
    item_variant?: string // Service duration/type
    location_id?: string
    price: number
    quantity: number
  }>
}

// Consent-aware tracking configuration
export interface TrackingConfig {
  enabled: boolean
  debug?: boolean
  anonymize_ip?: boolean
  cookie_domain?: string
  cookie_expires?: number
  cookie_prefix?: string
  user_id?: string
  custom_dimensions?: CustomDimensions
  custom_metrics?: CustomMetrics
  enhanced_ecommerce?: boolean
  cross_domain_tracking?: string[]
  excluded_referrers?: string[]
  sample_rate?: number // 0-100
}

// Tracking state interface
export interface TrackingState {
  initialized: boolean
  consent_given: boolean
  analytics_enabled: boolean
  marketing_enabled: boolean
  last_event?: {
    type: ConversionEventType
    timestamp: string
    params?: any
  }
  session: {
    id: string
    start_time: string
    events_count: number
    page_views: number
  }
  user: {
    id?: string
    type?: string
    first_seen: string
    last_seen: string
    total_events: number
    total_conversions: number
    total_value: number
  }
}